packages/react/src/components/login/components/WithAuthenticatedUser.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { isValidElement } from 'react';
+
+import { WithAuthentication } from './WithAuthentication';
+import { User } from '../types';
+import { getChildrenAsArray } from '../../../utils/getChildren';
+
+type Props = { user: User };
+
+export function createChildrenWithUser(children: React.ReactNode | null, user: User): React.ReactNode[] {
+  const renderChild = (child: React.FunctionComponentElement<unknown> | React.ReactElement, key: string) => {
+    return React.cloneElement(child, { key, ...child.props, user });
+  };
+  return getChildrenAsArray(children).map((child, index) => {
+    if (!child) {
+      return null;
+    }
+    const childType = typeof child;
+    if (childType === 'string' || childType === 'number') {
+      return child;
+    }
+    if (childType === 'function') {
```

Comment:
Should this have validity check as well? `if (childType === 'function' && isValidElement(child) {`

Reply:
If typeof child === "function", it will always pass the "isValidElement" -check too, I think?

Reply:
Not always. Consult the [isValidElement docs](https://react.dev/reference/react/isValidElement#caveats). I think the check would make this safer.

Reply:
Docs: 

> **Caveats** 
> Only [JSX tags](https://react.dev/learn/writing-markup-with-jsx) and objects returned by [createElement](https://react.dev/reference/react/createElement) are considered to be React elements.

Does that mean that a function cannot be a valid element and 
`if (childType === 'function' && isValidElement(child)` is always false? The docs have no examples with functions, so I can't tell. 

If `isValidElement` is added, tests fail because

```
<WithAuthenticatedUser>
  {({ user }) => {
  return <p id={elementIds.userName}>{user.profile.name}</p>;
  }}
  <span id={elementIds.hello}>Hello there!</span>
</WithAuthenticatedUser>
```

is not rendered anymore. 

Actually the component does not directly render the child when it is a function. The function is called with a user object and the result is cloned with `React.cloneElement` and then it is rendered. Perhaps that is safe enough? Or what is the scenario the `isValidElement` filters out?

Reply:
> Or what is the scenario the isValidElement filters out?

Usually TypeScript has given me errors when using `React.cloneElement` without checking for validity first, so I reckoned that it's always good to before cloning just so there are no errors.

> Actually the component does not directly render the child when it is a function. The function is called with a user object and the result is cloned with React.cloneElement and then it is rendered. Perhaps that is safe enough? 

I do consider this safe enough 🤔 I think you can leave the ìsValidElement` check out then.